reflectivity_ui/interfaces/data_handling/data_manipulation.py

```diff
@@ -160,33 +160,49 @@ def stitch_reflectivity(reduction_list, xs=None, normalize_to_unity=True, q_cuto
     return scaling_factors
 
 
-def _prepare_workspace_for_stitching(cross_section, ws_name):
+def _prepare_workspace_for_stitching(cross_sections, xs_input, global_fit, ws_name):
     """
     Create a workspace from a CrossSectionData object that we
     can call Stitch1D on.
-    :param CrossSectionData cross_section: cross section data object
+    :param dict cross_sections: dictionary with cross-section name and CrossSectionData objects
+    :param str xs: cross-section to use
+    :param bool global_fit: if True, merge data from all cross-sections
+    :param str ws_name: output workspace name
     """
-    n_total = len(cross_section.q)
-    p_0 = cross_section.configuration.cut_first_n_points
-    p_n = n_total - cross_section.configuration.cut_last_n_points
-    ws = api.CreateWorkspace(
-        DataX=cross_section.q[p_0:p_n],
-        DataY=cross_section._r[p_0:p_n],
-        DataE=cross_section._dr[p_0:p_n],
-        OutputWorkspace=ws_name,
-    )
+    if global_fit:
+        xs_names = cross_sections.keys()
+    else:
+        xs_names = [xs_input]
+
+    ws_list = []
+    for xs in xs_names:
+        cross_section = cross_sections[xs]
+        n_total = len(cross_section.q)
+        p_0 = cross_section.configuration.cut_first_n_points
+        p_n = n_total - cross_section.configuration.cut_last_n_points
+        ws_xs = api.CreateWorkspace(
+            DataX=cross_section.q[p_0:p_n],
+            DataY=cross_section._r[p_0:p_n],
+            DataE=cross_section._dr[p_0:p_n],
+            OutputWorkspace=xs,
+        )
+        ws_list.append(str(ws_xs))
+
+    ws = api.MergeRuns(ws_list)
     ws.setDistribution(True)
     ws = api.ConvertToHistogram(ws, OutputWorkspace=ws_name)
     return ws
```

Comment:
what happens with all the workspaces created in the previous lines with algorithm `CreateWorkspace`? Are they used in the future? If they're just temporary, they should be deleted with `DeleteWorkspaces`.

Reply:
I think there should only be four lingering workspaces ("On_On", "On_Off", ...) that will be reused every time this is run, but you're right, it is cleaner to delete the temporary workspaces. Hmm, and maybe I will give them more specific names (e.g. "tmp_prepare_for_stitching_On_On") in case the cross-sections names are used for workspaces somewhere else...